### PR TITLE
adds quota manager, which helps in updating the project quotas

### PR DESCRIPTION
**Pull request by yunus**
Provides a simple module quota.py to update the service quotas of projects programmatically via APIs.
The other option to update the quotas are via console which is a manual work. So this module enables GCP customer to automate the quota update process.

### CODE REVIEW

**Comment by dvmorris** [comment]
This is awesome! Have you ever tried updating a compute engine quota? I tried this morning and I am getting this error:

```
03/04/2020 07:28:05 INFO:URL being requested: POST https://serviceusage.googleapis.com/v1beta1/projects/*******/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%252Fnvidia_t4_gpus
/limits/%252Fproject%252Fregion/consumerOverrides?force=true&alt=json
Traceback (most recent call last):
  File "./update-computeengine-quota.py", line 109, in <module>
    project_quota=namespace.project_quota)
  File "./update-computeengine-quota.py", line 95, in main
    quota_updater.update_quota(project_quota, quota_dimensions)
  File "/home/dave/professional-services/tools/quota-manager/quota.py", line 171, in update_quota
    operation = self.__create_new_quota(new_quota, dimensions)
  File "/home/dave/professional-services/tools/quota-manager/quota.py", line 130, in __create_new_quota
    body=body)\
  File "/home/dave/.local/share/virtualenvs/quota-manager-7vDkce0R/lib/python3.7/site-packages/googleapiclient/_helpers.py", line 134, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/dave/.local/share/virtualenvs/quota-manager-7vDkce0R/lib/python3.7/site-packages/googleapiclient/http.py", line 898, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://serviceusage.googleapis.com/v1beta1/projects/*******/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%252F
nvidia_t4_gpus/limits/%252Fproject%252Fregion/consumerOverrides?force=true&alt=json returned "Precise limit override on metric 'compute.googleapis.com/nvidia_t4_gpus' with limit /project/region can't be applied in
 service :compute.googleapis.com">
```

I created a new class that passes the "dimension" attribute, and I tried updating a regional quota like nvidia_t4_gpus. I got the idea for the dimension attribute from this page: https://cloud.google.com/service-usage/docs/reference/rest/v1beta1/services.consumerQuotaMetrics.limits.adminOverrides#QuotaOverride

**Comment by yunus** [comment]
@dvmorris I have added the dimensions but still getting the same error as you have. I suspect that compute engine apis cannot be updated via APIs. I have asked about this internally. You are in CC.

**Comment by yunus** [comment]
@dvmorris it turns out that compute engine quotas cannot be updated via APIs yet. But soon they will be ready for update.

File: tools/quota-manager/Pipfile.lock [hunk 1]
```diff
@@ -0,0 +1,216 @@
+{
```
Comment (boredabdel) [inline tools/quota-manager/Pipfile.lock:1]: Pretty much looks good but this file is not needed, it's generated by pipenv for your installation, can you add a .gitignore file with this file in it and amend the changes please ? Ones done i will approve this.

Comment (yunus) [reply tools/quota-manager/Pipfile.lock:1]: Lock ensures the integrity. If someone hits a compromised pip server, sha digests in the lock are used to detect it.  See [the advise](https://pipenv-fork.readthedocs.io/en/latest/basics.html#general-recommendations-version-control)

can I keep it?

Comment (boredabdel) [reply tools/quota-manager/Pipfile.lock:1]: SGTM

**Comment by boredabdel** [comment]
I'm not able to merge this change but it looks good. Added Ryan and Jacob who i believe can merge

**Pull request merged**